src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -624,6 +662,25 @@ private bool CheckFieldValueKind(SyntaxNode node, BoundFieldAccess fieldAccess,
             return CheckIsValidReceiverForVariable(node, fieldAccess.ReceiverOpt, valueKind, diagnostics);
         }
 
+        private static bool CheckSimpleAssignmentValueKind(SyntaxNode node, BoundAssignmentOperator assignment, BindValueKind valueKind, DiagnosticBag diagnostics)
+        {
+            // Only ref-assigns produce LValues
+            if (assignment.RefKind != RefKind.None)
+            {
+                if (RequiresRefAssignableVariable(valueKind))
+                {
+                    return true;
+                }
+                if (RequiresAssignableVariable(valueKind))
+                {
+                    return assignment.RefKind != RefKind.RefReadOnly;
```

Comment:
I didn't understand this case. Is it possible to have `assignment.RefKind == RefKind.RefReadOnly`?

Reply:
Sure -- assigning a ref-readonly variable with have an assignment RefKind of RefReadOnly. This is an error because the left hand side must be assignable, and if the assignment RefKind is RefReadonly that means the left hand side must be readonly, i.e. not assignable.